TeXLive

fosslinux opened the pull request:
I have been working on this for a while now.. it is still not quite complete, but is getting there.

pdflatex works.
I have removed conflicting files from one of the packages.

Still TODO for runtime tests:

- [x] luatex
- [x] luajittex
- [x] xetex
- [x] bibtex
- [x] dvi*
- [x] pythontex
- [x] various utilities
- [x] variety of fonts/packages from each texmf package

Still TODO for building:

- [x] LuaJIT needs fixing on ppc64. @q66 will do that
- [ ] ???
- [x] `nostrip` and `nodebug` for texmf packages to speed up build times?

fosslinux commented:
Haven't tested building all of the data packages again yet... going to do that now. Tested `texlive-core` and it seems sane.

fosslinux commented:
Whoops, didn't mean to click that..

Chocimier commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+		rm -rf texmf-dist/
+	fi
+	rm -f LICENSE*
+	if [ -f "${FILESDIR}/duplicates.txt" ] ; then
```
Files will become outdated quite fast. Is it possible to compute list here?

fosslinux replied:
Not really, each file needs to be looked at on a case-by-case basis. It's annoying but necessary. I made a script that checks this. Should I include this somewhere in one of the `files/` directories?

Chocimier replied:
Yes it is preferred to share tools used to manage templates. Show it, and i'll try to make it part of build.

Chocimier commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+# python_version isn't needed for everything either
+python_version=3
+
+do_install() {
```
Helpers are meant to be composable with other helpers and build styles. Change to build style or make it function called from template's do_install.

If build style, extraction can work with create_wrksrc set by default, dafault do_extract and moving files in hook.

fosslinux replied:
Hm, ok, I think I might change it to a build-style, should simplify the code somewhat further.

fosslinux replied:
Moved to build-style in latest push.

Chocimier commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+do_install() {
+	mkdir -p "${PKGDESTDIR}/usr/share/texmf-dist"
+	unzip "${pkgname}-${version}-src.zip"
+	find . -maxdepth 1 -print -name "*.tar.xz" \
```
Really only xz?

fosslinux replied:
Yes, inside the zip are a bunch of .tar.xz files...

Chocimier commented on srcpkgs/texlive-bibtexextra/template:
```diff
@@ -0,0 +1,14 @@
+maintainer="fosslinux <fosslinux@aussies.space>"
+license="GPL-2.0-or-later"
+homepage="http://tug.org/texlive/"
+distfiles="https://sources.archlinux.org/other/texlive/${pkgname}-${version}-src.zip"
```
No upstream tarballs?

fosslinux replied:
Upstream does not release tarballs, as it is designed to be used with `tlmgr`. This is what alpine uses too, and what @q66 said to use.

fosslinux replied:
It is not super difficult to generate our own tarballs if needed, but I don't see any reason to create our own tarballs for as long as archlinux provides tarballs.

Chocimier replied:
ok

fosslinux commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+		-exec bsdtar -C "${PKGDESTDIR}/usr/share/texmf-dist" -xf {} \;
+	cd "${PKGDESTDIR}/usr/share/texmf-dist/"
+	if [ -d "texmf-dist" ] ; then
+		rsync -ar texmf-dist/ ./
```
No, it is not - directories need to be merged. This is not possible with mv, so rsync is used. It's not a very large dependency and makes the code much cleaner.

Chocimier replied:
perfect

fosslinux commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+		rsync -ar texmf-dist/ ./
+		rm -rf texmf-dist/
+	fi
+	rm -f LICENSE*
```
Some of the packages have LICENSE in `/usr/share/texmf-dist` directly. They do not need to be there.

fosslinux commented:
@Chocimier here is the script i have been using to check for duplicates.

```
#!/bin/bash
for i in srcpkgs/texlive* ; do
    case ${i} in *-bin) ;;
        *)
            if [ $(basename ${i}) = $1 ] ; then
                continue
            fi
            echo $i
            comm -12 <(xbps-query --repository=hostdir/binpkgs/$(git branch --show-current) -i -f $(basename ${i})) <(xbps-query --repository=hostdir/binpkgs/$(git branch --show-current) -i -f $1) 2>/dev/null
            ;;
    esac
done
```

Please note for each duplicate there is not a one size fits all solution. Each needs to be manually human reviewed and decided which package each duplicate actually fits into...

q66 commented:
We can probably use a single `duplicates.txt` file and straight take it from Alpine, instead of having many scattered around and hand-checked, then shove the file in some appropriate location.

fosslinux commented:
Hm, yes, I guess they could all be merged.

fosslinux commented on srcpkgs/pygments/template:
```diff
@@ -0,0 +1,17 @@
+# Template file for 'pygments'
+pkgname=pygments
```
Fixed in latest push (oops noticed typo, fixing that in next push)

fosslinux commented on common/build-helper/texmf.sh:
```diff
@@ -0,0 +1,25 @@
+
+do_install() {
+	mkdir -p "${PKGDESTDIR}/usr/share/texmf-dist"
+	unzip "${pkgname}-${version}-src.zip"
```
Done by moving to default `do_extract`.

q66 commented on common/environment/build-style/texmf.sh:
```diff
@@ -0,0 +1,5 @@
+# rsync isn't needed for everything but it's far easier to just put it here
+hostmakedepends+=" xz rsync"
```
Don't need xz

q66 commented on srcpkgs/texlive/template:
```diff
@@ -0,0 +1,346 @@
+ harfbuzz-devel icu-devel libpaper-devel libpng-devel mpfr-devel
+ poppler-devel pixman-devel libteckit-devel zlib-devel zziplib-devel
+ libXaw-devel"
+depends="dialog ghostscript openjdk-jre perl-Tk"
```
Should depend on texlive-core

fosslinux replied:
I'm still not entirely convinced on that. The texmf distribution seems to be somewhat more isolated from the texlive binaries than I previously thought. If someone developed an alternative texmf distribution, then there would be two texmf options, and you could use either... I was thinking to just depend on texlive-core in each of the metapackages.

Feel free to correct me, I'm not sure my reasoning is very logical, and since there is only really one texmf distribution now, maybe your suggestion makes sense.

q66 replied:
nobody's gonna develop another distribution, and the texlive package literally needs some perl scripts from texlive-core to work at all

q66 commented on srcpkgs/texlive/files/patches/ppc/musl-ppc-secureplt.patch:
```diff
@@ -0,0 +1,93 @@
+Imported from https://github.com/LuaJIT/LuaJIT/pull/486.
```
Don't copy luajit patches, just apply them from luajit

q66 commented on srcpkgs/texlive/template:
```diff
@@ -0,0 +1,346 @@
+# Template file for 'texlive'
```
Add metapackages for these various sets

fosslinux replied:
Will do next

Chocimier commented:
> Each needs to be manually human reviewed and decided which package each duplicate actually fits into...

In that case, let's have a file containing pairs of path and package that should contain it. If package do not longer contain file build fails, for all other packages, if present, file will be removed and recorded in package. This allows to fix on deduplication upstream.

For new duplicates, maybe let's have script like below in repo to be run on updates, or equivalent as do_check of texmf and pre_check of texlive?

	#!/bin/bash

	declare -A files
	exitcode=0

	for i in srcpkgs/texlive* ; do
		pkg="$(basename $i)"
		case ${i} in
			*-bin) ;;
			*)
				for file in $(xbps-query
						--repository=hostdir/binpkgs/$(git branch --show-current)
						-f $pkg | sed 's/ -> .*//'); do
					if [ -n "${files[$file]}" ]; then
						echo "$file present in both $pkg and ${files[$file]}"
						exitcode=1
					else
						files[$file]=$pkg
					fi
				done
				;;
		esac
	done

	exit $exitcode

fosslinux commented:
I like this a lot, this is very clean. That is a great idea to put that in `do_check`/`pre_check`. That would make it much easier and simpler!

> let's have a file containing pairs of path and package

Do you mean, have a file somewhere in common with something like

```
 /usr/share/texmf-dist/scripts/crossrefware/bibmradd.pl texlive-bibtexextra
```
etc etc, replacing the current duplicates.txt? I really like that :)

or even,
```
scripts/crossrefware/bibmradd.pl texlive-bibtexextra
```

because it's only in /usr/share/texmf-dist where there will ever be duplicates.

fosslinux commented:
@q66 PPC status:

- `ppc` builds
- `ppc64` and `ppc64le` fail with:

```
buildvm_ppc.dasc: Assembler messages:
buildvm_ppc.dasc:388: Error: syntax error; found `@', expected `,'
buildvm_ppc.dasc:388: Error: junk at end of line: `@plt'
```

q66 commented:
Apply fixes from here: https://gist.github.com/q66/3069e8117d5e05fa933880a0b8e26723

Changes:

1) removed unused files
2) texluajit buildsystem now recognizes our ppc64 targets (no longer errors, tested that)
3) dropped aarch64 using arm host-cc - this is wrong, there is no requirement for aarch64 to be able to execute 32-bit stuff and a lot of systems can't (it's effectively a different architecture)
4) added ppc64 BE being able to use 32-bit ppc compiler for luajit cross (not LE as we don't have ppcle yet)
5) fixed formatting
6) all patches from luajit are now applied, not just ppc ones
7) made that weird luajit configure block cross-only, and fixed triplet for target

i haven't tested building whole texlive yet, so do that yourself

fosslinux commented:
3. Oh, I was not aware of that, thanks
4. Thanks, good
7. Oops, missed over that one, looks good! That weird luajit configure block is because luajit is stupid for cross, as you would know yourself from the LuaJIT template, as it requires a bunch of nonstandard variables :sigh:

fosslinux commented:
Uh, didn't mean to push that, the patch is subtly broken in some ways that still need fixing.

q66 commented:
We still have some dupes:

```
ERROR: texlive-pstricks-2020.55289_1: file `/usr/share/texmf-dist/scripts/pedigree-perl/pedigree.pl' already installed by package texlive-20200406_1.
ERROR: texlive-pstricks-2020.55289_1: file `/usr/share/texmf-dist/scripts/pst-pdf/ps4pdf' already installed by package texlive-20200406_1.
ERROR: texlive-pstricks-2020.55289_1: file `/usr/share/texmf-dist/scripts/pst2pdf/pst2pdf.pl' already installed by package texlive-20200406_1.
ERROR: texlive-langjapanese-2020.55320_1: file `/usr/share/texmf-dist/scripts/ptex2pdf/ptex2pdf.lua' already installed by package texlive-LuaTeX-20200406_1.
```

Also getting an error:

```
Can't locate TeXLive/TLUtils.pm in @INC (you may need to install the TeXLive::TLUtils module) (@INC contains: /usr/share/tlpkg /usr/share/texmf-dist/scripts/texlive /usr/lib/perl5/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/core_perl /usr/share/perl5/core_perl .) at /bin/mktexfmt line 46.
BEGIN failed--compilation aborted at /bin/mktexfmt line 46.
```

I could work that around by symlinking `/usr/share/texmf-dist/tlpkg/TeXLive` to `/usr/share/texmf-dist/scripts/texlive` but that's probably not correct (but it would make a pdf afterwards)

q66 commented:
Looks like alpine's `texlive` package contains

```
/usr/share/texmf-dist/scripts/texlive/TeXLive/TLConfig.pm 
/usr/share/texmf-dist/scripts/texlive/TeXLive/TLUtils.pm
```

and ours does not.

q66 commented:
Fixed:

```
commit 1e3dfc39137373c30e1aa27d7d6c0c7c0dec4da3
Author: q66 <daniel@octaforge.org>
Date:   Mon Jun 29 22:03:47 2020 +0200

    fixes from alpine

diff --git a/srcpkgs/texlive/template b/srcpkgs/texlive/template
index c254de4464..25d9c749d5 100644
--- a/srcpkgs/texlive/template
+++ b/srcpkgs/texlive/template
@@ -221,6 +221,8 @@ EOF
 }
 
 post_install() {
+	cp -rf ${wrksrc}/texk/tests/TeXLive \
+		"${DESTDIR}/usr/share/texmf-dist/scripts/texlive"
 	sed -i -e 's:^\(TEXMFROOT *= *\)$SELFAUTOPARENT$:\1/usr/share:g' \
 		"${DESTDIR}/usr/share/texmf-dist/web2c/texmf.cnf"
 	# The user should not use tlmgr
@@ -233,8 +235,10 @@ post_install() {
 	for cmd in amstex cslatex csplain eplain etex jadetex latex lollipop mex \
 				llatex mltex pdfetex pdfcslatex pdfcsplain pdfjadetex pdflatex \
 				pdfmex pdfxmltex texsis utf8mex xmltex ; do
-		ln -s pdftex "${PKGDESTDIR}/usr/bin/${cmd}"
+		ln -s pdftex "${DESTDIR}/usr/bin/${cmd}"
 	done
+	ln -s eptex "${DESTDIR}/usr/bin/platex"
+	ln -s euptex "${DESTDIR}/usr/bin/uplatex"
 }
 
 texlive-XeTeX_package() {
```

q66 commented:
Also need this:

```
commit e5230d2ff22226b8061625e708db8f4c2c85919f
Author: q66 <daniel@octaforge.org>
Date:   Mon Jun 29 22:49:35 2020 +0200

    fix luajittex on aarch64, ppc64

diff --git a/srcpkgs/texlive/patches/luajit-elfv2.patch b/srcpkgs/texlive/patches/luajit-arch-fixes.patch
similarity index 53%
rename from srcpkgs/texlive/patches/luajit-elfv2.patch
rename to srcpkgs/texlive/patches/luajit-arch-fixes.patch
index eac6ddb2be..ae1e95c04c 100644
--- a/srcpkgs/texlive/patches/luajit-elfv2.patch
+++ b/srcpkgs/texlive/patches/luajit-arch-fixes.patch
@@ -1,8 +1,21 @@
 Have texluajit recognize ppc64 ELFv2 ABI and therefore emit correct dynasm
 stuff.
 
+Also, fix luajittex crashing on non-x86 64-bit systems (MAP_32BIT only exists
+on x86 and defining it on other systems will activate a codepath in luajit
+that segfaults).
+
 --- libs/luajit/configure
 +++ libs/luajit/configure
+@@ -14243,7 +14243,7 @@ else
+ fi
+ if grep 'LJ_ARCH_BITS 64' conftest.i >/dev/null 2>&1; then :
+   echo '-D P64' >>dynasm_flags
+-         if test "x$LJHOST" = xLinux; then :
++         if test "x$LJHOST" = xLinux -a "x$LJARCH" = xx64; then :
+   LUAJIT_DEFINES="$LUAJIT_DEFINES -DMAP_32BIT=0x40"
+ fi
+ fi
 @@ -14294,6 +14294,9 @@ fi
  fi
                  if grep 'LJ_ARCH_PPC32ON64 1' conftest.i >/dev/null 2>&1; then :
@@ -15,6 +28,15 @@ stuff.
    DASM_ARCH=ppc64
 --- libs/luajit/m4/lj-system.m4
 +++ libs/luajit/m4/lj-system.m4
+@@ -52,7 +52,7 @@ AS_IF([grep 'LJ_LE' conftest.i >/dev/null 2>&1],
+        [echo '-D ENDIAN_BE' >>dynasm_flags])
+ AS_IF([grep 'LJ_ARCH_BITS 64' conftest.i >/dev/null 2>&1],
+         [echo '-D P64' >>dynasm_flags
+-         AS_IF([test "x$LJHOST" = xLinux],
++         AS_IF([test "x$LJHOST" = xLinux -a "x$LJARCH" = xx64],
+                [LUAJIT_DEFINES="$LUAJIT_DEFINES -DMAP_32BIT=0x40"])])
+ AS_IF([grep 'LJ_HASJIT 1' conftest.i >/dev/null 2>&1],
+         [echo '-D JIT' >>dynasm_flags])
 @@ -84,6 +84,8 @@ AS_CASE([$LJARCH],
                        [echo '-D ROUND' >>dynasm_flags])
                  AS_IF([grep 'LJ_ARCH_PPC32ON64 1' conftest.i >/dev/null 2>&1],

```

fosslinux commented:
Thanks for those. Will be applied in next push (coming as soon as i test these other changes).

### Architecture master list

**Building:**

- [x] x86_64
- [x] x86_64-musl
- [x] i686
- [x] i686-musl
- [x] aarch64
- [x] aarch64-musl
- [x] armv7l
- [x] armv7l-musl
- [x] armv6l
- [x] armv6l-musl
- [x] armv5tel
- [x] armv5tel-musl
- [x] ppc
- [ ] ppc-musl (not sure for ppc*-musl, should but will check anyway)
- [x] ppc64
- [ ] ppc64-musl
- [x] ppc64le
- [ ] ppc64le-musl
- [ ] mips-musl (needs java removal, in next push)
- [ ] mipshf-musl
- [ ] mipsel-musl
- [ ] mipselhf-musl

**Running:**

- [x] x86_64
- [x] x86_64-musl
- [x] i686
- [ ] aarch64
- [ ] aarch64-musl
- [ ] armv7l
- [ ] armv7l-musl
- [ ] armv6l
- [ ] armv6l-musl
- [ ] ppc - I will leave it to q66's discretion as to what ppc's need runtime testing
- [ ] ppc-musl
- [ ] ppc64
- [ ] ppc64-musl
- [x] ppc64le
- [ ] ppc64le-musl

@pullmoll If you would like to runtime test on i686-musl, armv5tel and mips*, be my guest, but no pressure

I will runtime test the ARMs

fosslinux commented:
WRT mips*, I am not sure how to approach the java issue - mips does not have java, but there are a number of jars.

In the texmf data packages, there are a few scattered jar files (we should look into rebuilding those, but unrelated), but they are noarch. mips* should have a java opt like I have added to texlive, but that isn't possible because they are noarch.

Suggestions?

q66 commented:
why would MIPS not have java? either way don't bother with mips, it doesn't have packages and nobody is gonna use texlive on that anyway

q66 commented:
btw, we should probably drop the removal of `tlmgr`, alpine does it but it's wrong, `tlmgr` can be used in user mode to install extra stuff even in a packaged distribution

fosslinux commented:
OpenJDK dosen't support mips (not sure why)

The java option was a workaround for mips - I'm not going to block merging this over mips, ofc. We can come back to it later if there is an easier way to do it. I'll just drop the option

RE: `tlmgr`. Hm, yes, I see the use for user mode. When this is documented in Void Docs, however, I think we need to make it clear that it is completely unsupported to use `tlmgr` for installing stuff onto the system.

q66 commented:
openjdk supports every architecture, via the zero backend, that's also how it's supported on ppc32 - it might not have a JIT, but that doesn't matter

anyway openjdk is not needed to use texlive so drop the dependency

fosslinux commented:
Well, the openjdk mips thing is a seperate problem so im not going to worry about it for this pr

Oh, so it is, I will remove the dependency.

Grepping for jars in texmf data packages results in 
```
texlive
texlive-bibtexextra
/usr/share/texmf-dist/scripts/bib2gls/bib2gls.jar
/usr/share/texmf-dist/scripts/bib2gls/convertgls2bib.jar
/usr/share/texmf-dist/scripts/bib2gls/texparserlib.jar
texlive-core
/usr/share/texmf-dist/scripts/arara/arara.jar
/usr/share/texmf-dist/scripts/ketcindy/ketjava/KetCindyPlugin.jar
/usr/share/texmf-dist/scripts/latex2nemeth/latex2nemeth-v1.0.2.jar
/usr/share/texmf-dist/scripts/texosquery/texosquery-jre5.jar
/usr/share/texmf-dist/scripts/texosquery/texosquery-jre8.jar
/usr/share/texmf-dist/scripts/texosquery/texosquery.jar
/usr/share/texmf-dist/scripts/texplate/texplate.jar
/usr/share/texmf-dist/tex4ht/bin/tex4ht.jar
texlive-fontsextra
texlive-formatsextra
texlive-games
texlive-humanities
texlive-langchinese
texlive-langcyrillic
texlive-langextra
texlive-langgreek
texlive-langjapanese
texlive-langkorean
texlive-latexextra
/usr/share/texmf-dist/scripts/pax/pax.jar
texlive-music
texlive-pictures
texlive-pstricks
texlive-publishers
texlive-science
```

I think that each of those with jars in them should have a java dependency then? Some of them seem to be reimplementations of something in perl though......

q66 commented:
no, don't add any java dependency *anywhere* - these jars are all either optional, supplementary, have a soft bail-out, etc

just don't bother with it, people who want to use them will install java themselves, the rest of the world doesn't have to care

fosslinux commented:
Ok nvm:

- arara needs java only https://github.com/cereda/arara
- pax needs java https://ctan.org/tex-archive/macros/latex/contrib/pax?lang=en
- bib2gls needs java https://ctan.org/tex-archive/support/bib2gls?lang=en (requirements)

Hence each of these pkgs needs java

fosslinux commented:
Fixed dupes temporarily, removed java crap, dropped removal of tlmgr; also made a patch to fix a check.

@Chocimier I believe is writing a do_check for the duplicates.

Next up: Metapacakges.

Chocimier commented:
@fosslinux: Following has no checks for duplicates in texlive because it would need to run after do_install and list is rough, but otherwise works.

https://github.com/Chocimier/void-packages-org/commits/texlive

fosslinux commented:
As far as I can remember, the only thing left to do now is:

- [ ] Final testing
- [ ] Fix the trigger

Anything else I can't remember?

q66 commented:
i think that's it

Chocimier commented:
Corrected removing duplicates from texlive. duplicates.txt are not used anymore. Could you check if ownership.txt is correct?

fosslinux commented:
Sorry, not sure how I did that last one. Fixed the trigger and broken on mips, say as so in the template. Fixed a few dupes too

fosslinux commented:
I have fixed symlinks and changed distfiles to over http because Travis dosen't seem to like ftp.... dont know why. Can change it back just before merge but I want to see travis give the green tick.

fosslinux commented:
I believe this is ready to ship. I have runtime tested on ARM (all good :)) and built every package for every arch.

@q66 @Chocimier anything further?

q66 commented:
trigger has broken indent

fosslinux commented:
Fixed merge conflict and fixed indent

Chocimier commented on Manual.md:
```diff
@@ -931,6 +932,8 @@ Environment variables for a specific `build_style` can be declared in a filename
 
     `common/environment/build-style/gnu-configure.sh`
+    
+- `texmf` For archlinux's texmf zip/tarballs that need to go into /usr/share/texmf-dist.
```
Archlinux isn't so important. Duplicates handling is more.

Chocimier commented on Manual.md:
```diff
@@ -1895,6 +1898,20 @@ Example: `transmission unprivileged user - for uninstalled package transmission`
+databases using `texhash` and `fmtutil-sys`, to add or remove any new hashes or
+formats.
+
+It runs on every package that changes /usr/share/texmf-dist. This is likely overkill,
```
Fine.

Chocimier commented on common/environment/build-style/texmf/ownership.txt:
```diff
@@ -0,0 +1,229 @@
+scripts/epspdf/epspdftk.tcl texlive
+scripts/epstopdf/epstopdf.pl texlive
+scripts/exceltex/exceltex texlive
+scripts/exceltex/exceltex texlive
```
Please run through `sort -u`

Chocimier commented on common/shlibs:
```diff
@@ -3978,3 +3978,7 @@ libsepol.so.1 libsepol-3.0_1
+libkpathsea.so.6 texlive-20200406_1
+libtexluajit.so.2 texlive-LuaTeX-20200406_1
+libtexlua53.so.5 texlive-LuaTeX-20200406_1
+libptexenc.so.1 texlive-20200406_1
```
Btw, adding entries few lines up prevent conflicts with majority of PRs that add at end.

Chocimier commented on srcpkgs/texlive-basic/template:
```diff
@@ -0,0 +1,18 @@
+license="GPL-2.0-or-later"
+homepage="https://www.tug.org/texlive/"
+
+depends="texlive>=20200406
```
Please move up to usual place.

Chocimier replied:
And fix xlint please.

Chocimier commented on srcpkgs/texlive-latexextra/template:
```diff
@@ -0,0 +1,14 @@
+homepage="http://tug.org/texlive/"
+distfiles="https://sources.archlinux.org/other/texlive/${pkgname}-${version}-src.zip"
+checksum=a2ee21a2a6f00def6bec620a85bb8116de97a16dcd98adaa308eb455fe0e7fcb
+python_version=3
```
Redundant?

Chocimier commented on srcpkgs/texlive-music/template:
```diff
@@ -0,0 +1,14 @@
+homepage="http://tug.org/texlive/"
+distfiles="https://sources.archlinux.org/other/texlive/${pkgname}-${version}-src.zip"
+checksum=47f418d851a0f838c08be406135ef338533e32cf0bc9c4ffefeaa0139973e8f1
+python_version=3
```
Redundant?

Chocimier commented on srcpkgs/texlive/template:
```diff
@@ -0,0 +1,388 @@
+ harfbuzz-devel icu-devel libpaper-devel libpng-devel mpfr-devel
+ poppler-devel pixman-devel libteckit-devel zlib-devel zziplib-devel
+ libXaw-devel"
+depends="dialog ghostscript perl-Tk texlive-core"
```
Should it depend on new version of triggers?

fosslinux replied:
I was discussing this with q66 and we thought no, but after doing a few more installations, it seems to be much more logical to depend on the triggers, otherwise if a user doesn't do `xbps-install -Su` before `xbps-install texlive-most` (or whatever) they'll have a broken install...

So yes I will make it depend on new triggers.

Chocimier commented on srcpkgs/xbps-triggers/files/texmf-dist:
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+#
+# Updates the shared-mime-info db file with update-mime-database(1).
```
Please adjust.

Chocimier commented on srcpkgs/xbps-triggers/files/texmf-dist:
```diff
@@ -0,0 +1,39 @@
+	echo "post-install post-remove"
+	;;
+run)
+	if [ ! -x ${texhash} ] && [ ! -x ${fmtutil} ]; then
```
This runs both when one exists.

Chocimier commented on srcpkgs/xbps-triggers/files/texmf-dist:
```diff
@@ -0,0 +1,39 @@
+	echo "Updating texmf-dist hashes..." 
+	${texhash} 2>&1 >/dev/null || true # silence strange errors
+	echo "Updating texmf-dist formats..."
+	${fmtutil} --all >/dev/null || true
```
With luatex-minimal, I get: *[ERROR]: not building luatex due to missing engine: luatex.*

Is it expected?

fosslinux replied:
Yes this is expected, because luatex is not installed, but its still in `fmtutil.cnf`. Maybe I should silence those errors? Or add a message at the beginning saying something like "Missing engine errors are due to missing packages"?

ericonr commented on Manual.md:
```diff
@@ -1895,6 +1898,20 @@ Example: `transmission unprivileged user - for uninstalled package transmission`
 
 This trigger can only be used by using the `system_accounts` variable.
 
+<a id="triggers_system_accounts"></a>
+#### texmf-dist
```
You can just use an achor for the link, `[texmf-dist](#texmf-dist)`. That way you help with #21609 :p

ericonr replied:
Unless of course there is an argument against moving to anchors.

fosslinux replied:
Should I, or not? I feel like this is something that should be standardized, so maybe its better to leave it for a sweeping-change PR...

Chocimier replied:
Keep consistent with what is in master.

Chocimier commented on srcpkgs/texlive/template:
```diff
@@ -0,0 +1,388 @@
+ harfbuzz-devel icu-devel libpaper-devel libpng-devel mpfr-devel
+ poppler-devel pixman-devel libteckit-devel zlib-devel zziplib-devel
+ libXaw-devel"
+depends="dialog ghostscript perl-Tk texlive-core xbps-triggers"
```
xbps-triggers>=0.115_1, otherwise update is not forced.

Chocimier commented on srcpkgs/xbps-triggers/files/texmf-dist:
```diff
@@ -0,0 +1,40 @@
+	;;
+run)
+	if [ -x ${texhash} ]; then
+    	echo "Updating texmf-dist hashes..." 
```
Indentation of whole `run` is broken.

Chocimier commented on srcpkgs/xbps-triggers/files/texmf-dist:
```diff
@@ -0,0 +1,40 @@
+    	echo "Updating texmf-dist formats..."
+	    ${fmtutil} --all >/dev/null || true
+    fi
+    exit 0
```
redundant?

fosslinux commented:
Addressed.

q66 commented:
requesting feedback from @Duncaen (since this will use space on the builders and mirrors) and possibly @leahneukirchen (since she maintains the other texlive stuff)

q66 commented:
Got an OK. I'll build the PR on my machine for some arches, test some more and we can merge it after

fosslinux commented:
Woohoo
